src/Layout.js

```diff
@@ -654,9 +663,112 @@ var computeLayout = (function() {
+      var/*float*/ currentLead = getPaddingAndBorder(node, leading[crossAxis]);
+
+      var/*css_align_t*/ alignContent = getAlignContent(node);
+      if (alignContent == CSS_ALIGN_FLEX_END) {
+        currentLead += remainingCrossDim;
+      }
+      else if (alignContent == CSS_ALIGN_CENTER) {
+        currentLead += remainingCrossDim / 2;
+      }
+      else if (alignContent == CSS_ALIGN_STRETCH) {
+        if (nodeCrossAxisInnerSize > linesCrossDim) {
+          crossDimAdd = (remainingCrossDim / linesCount);
+        }
+      }
+
+      // find the first node on the first line
+      for (i = 0; i < node.children.length; ) {
+        var/*int*/ startIndex = i;
+        var/*int*/ lineIndex = -1;
+
+        // get the first child on the current line
+        {
+          child = node.children[i];
+          if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+            ++i;
+            continue;
+          }
+          lineIndex = child.lineIndex;
+        }
+
+        // compute the line's height and find the endIndex
+        var/*float*/ lineHeight = 0;
+        for (ii = startIndex; ii < node.children.length; ++ii) {
+          child = node.children[ii];
+          if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+            continue;
+          }
+          if (child.lineIndex != lineIndex) {
+            break;
+          }
+          if (!isUndefined(child.layout[dim[crossAxis]])) {
+            lineHeight = fmaxf(lineHeight,child.layout[dim[crossAxis]] +
+                               getMarginAxis(child,crossAxis));
+          }
+        }
+        var/*int*/ endIndex = ii;
+        lineHeight += crossDimAdd;
+
+        for (ii = startIndex; ii < endIndex; ++ii) {
+          child = node.children[ii];
+          if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+            continue;
+          }
+
+          var/*css_align_t*/ alignItem = getAlignItem(node, child);
+          var/*float*/ crossPosition = child.layout[pos[crossAxis]]; // preserve current position if someting goes wrong with alignItem?
+          if (alignItem == CSS_ALIGN_FLEX_START) {
+            crossPosition = currentLead + getMargin(child,leading[crossAxis]);
+          }
+          else if (alignItem == CSS_ALIGN_FLEX_END) {
+            crossPosition = currentLead + lineHeight -
+              getMargin(child,trailing[crossAxis]) -
+              child.layout[dim[crossAxis]];
+          }
+          else if (alignItem == CSS_ALIGN_CENTER) {
+            var/*float*/ childHeight = child.layout[dim[crossAxis]];
+            crossPosition = currentLead + ((lineHeight - childHeight)/2);
+          }
+          else if (alignItem == CSS_ALIGN_STRETCH) {
+            crossPosition = currentLead + getMargin(child,leading[crossAxis]);
+            // TODO: Correctly set the height of items with undefined (auto)
+            //       crossAxis dimension.
+          }
+          child.layout[pos[crossAxis]] = crossPosition;
```

Comment:
since you always assign to child.layout[pos[crossAxis]], do you really need the variable? Can you just assign it in every branch of the if/